crates/task-impls/src/consensus/mod.rs

```diff
@@ -829,6 +860,46 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
                     };
                 }
             }
+            HotShotEvent::QuorumVoteSend(vote) => {
+                let Some(proposal) = self.current_proposal.clone() else {
+                    return;
+                };
+                let new_view = proposal.get_view_number() + 1;
+                // In future we can use the mempool model where we fetch the proposal if we don't have it, instead of having to wait for it here
+                // This is for the case where we form a QC but have not yet seen the previous proposal ourselves
+                let should_propose = self.quorum_membership.get_leader(new_view) == self.public_key
+                    && self.consensus.read().await.high_qc.view_number
+                        == proposal.get_view_number();
+
+                if should_propose {
+                    debug!(
+                        "Attempting to publish proposal after voting; now in view: {}",
+                        *new_view
+                    );
+                    let _ = self.publish_proposal(new_view, event_stream.clone()).await;
```

Comment:
I tried to debug the proposal task tests and got a compilation error at this line. I think it's because `publish_proposal` is gated by `not(feature = "dependency-tasks")` but this `QuorumVoteSend` event handling code isn't.

Reply:
This can be fixed by adding `#[cfg(not(feature = "dependency-tasks"))]` to `HotShotEvent::QuorumVoteSend(vote) => {`. I don't think this is needed for dependency tasks.